apache/zeppelin
# [ZEPPELIN-2439] Use babel-preset-env to specify browser compatibility explicitly

### What is this PR for?

Use babel-preset-env to specify browser compatibility explicitly.

- This will help to resolve some IE browser issues.
- Additionally, we don't need to add more presets like es2017, just update preset env when we need.

Specified browser lists are

- last 5 versions of major browsers
- `> %5` in the world
- IE 9+
- edge 12+
- safari 7+
- chrome 47+
- firefox 31+ (due to travis CI)

See also - [Browser Compatibility Table](https://kangax.github.io/compat-table/es6/)

### What type of PR is it?
[Improvement]

### Todos
* [x] - Setup babel-preset-env for web
* [x] - Setup babel-preset-env for helium

### What is the Jira issue?

[ZEPPELIN-2439](https://issues.apache.org/jira/browse/ZEPPELIN-2439)

### How should this be tested?

1. should be able to bundle existing helium packages
2. execute `npm run build` in `zeppelin-web`
3. should be able to see message like

```
babel-preset-env: `DEBUG` option

Using targets:
{
  "chrome": 47,
  "edge": 12,
  "firefox": 31,
  "ie": 7,
  "ios": 8,
  "safari": 7
}

Modules transform: commonjs

Using plugins:
  check-es2015-constants {"chrome":47,"edge":12,"firefox":31,"ie":7,"ios":8,"safari":7}
  transform-es2015-arrow-functions {"edge":12,"firefox":31,"ie":7,"ios":8,"safari":7}
  transform-es2015-block-scoped-functions {"firefox":31,"ie":7,"ios":8,"safari":7}
  transform-es2015-block-scoping {"chrome":47,"edge":12,"firefox":31,"ie":7,"ios":8,"safari":7}
  transform-es2015-classes {"edge":12,"firefox":31,"ie":7,"ios":8,"safari":7}
  transform-es2015-computed-properties {"firefox":31,"ie":7,"safari":7}
  transform-es2015-destructuring {"chrome":47,"edge":12,"firefox":31,"ie":7,"ios":8,"safari":7}
  transform-es2015-duplicate-keys {"firefox":31,"ie":7,"ios":8,"safari":7}
  transform-es2015-for-of {"chrome":47,"edge":12,"firefox":31,"ie":7,"ios":8,"safari":7}
  transform-es2015-function-name {"chrome":47,"edge":12,"firefox":31,"ie":7,"ios":8,"safari":7}
  transform-es2015-literals {"firefox":31,"ie":7,"ios":8,"safari":7}
  transform-es2015-object-super {"edge":12,"firefox":31,"ie":7,"ios":8,"safari":7}
  transform-es2015-parameters {"chrome":47,"edge":12,"firefox":31,"ie":7,"ios":8,"safari":7}
  transform-es2015-shorthand-properties {"firefox":31,"ie":7,"ios":8,"safari":7}
  transform-es2015-spread {"edge":12,"firefox":31,"ie":7,"ios":8,"safari":7}
  transform-es2015-sticky-regex {"chrome":47,"edge":12,"ie":7,"ios":8,"safari":7}
  transform-es2015-template-literals {"edge":12,"firefox":31,"ie":7,"ios":8,"safari":7}
  transform-es2015-typeof-symbol {"firefox":31,"ie":7,"ios":8,"safari":7}
  transform-es2015-unicode-regex {"chrome":47,"edge":12,"firefox":31,"ie":7,"ios":8,"safari":7}
  transform-regenerator {"chrome":47,"edge":12,"firefox":31,"ie":7,"ios":8,"safari":7}
  transform-exponentiation-operator {"chrome":47,"edge":12,"firefox":31,"ie":7,"ios":8,"safari":7}
  transform-async-to-generator {"chrome":47,"edge":12,"firefox":31,"ie":7,"ios":8,"safari":7}
  syntax-trailing-function-commas {"chrome":47,"edge":12,"firefox":31,"ie":7,"ios":8,"safari":7}
```


### Screenshots (if appropriate)

NONE

### Questions:
* Does the licenses files need update? - NO
* Is there breaking changes for older versions? - NO
* Does this needs documentation? - NO
